Feature_time_series_db_cleanup/Feature_time_series_db_cleanup.md

```diff
@@ -0,0 +1,29 @@
+# Feature: Time series database cleanup
+
+## Problem
+
+Omotes 'forgets' about jobs when they are completed and the results have been retrieved.
+Except for the time series data which are stored in a database, and the results are kept
+indefinitely. This will cause the database to grow without an option to clean up, since omotes does
+not know which data is still relevant.
+Most front-ends will have an option to delete omotes runs which should also remove the time series
+data. Currently, this is possible by directly accessing the times series database. This is a bit
+cumbersome since every frontend will need to implement this, knowledge of the database structure is
+needed, and it will be database type dependent.
+
+## Proposal
+
+If omotes deals with removing the time series data, only omotes code needs to be updated in the case
+of:
+
+- an update of the database structure
+- use of a different type of time series database
+
+The proposal is to create a `job_deletions` queue to which an SDK can publish. The orchestrator will

```

Comment:
The format of these messages are going to be key. I propose we use the ESDL id as that is going to be the identifier in the timeseries database on which we can select timeseries data. An OMOTES Job_id is not available in the timeseries db.

Reply:
How are we dealing with the same ESDL (same id) being submitted for OMOTES jobs many times? For instance, if the user changes ESDL a few times and then runs OMOTES jobs respectively, or if the same ESDL is shared and run by different users. Should both ESDL id and job_name, (also job run creator?) be used to identify which timeseries data to be deleted?

Also, do we need to make sure timeseries data can only be deleted by authorized users? For instance, job run creator or NWN admin

Reply:
How are we dealing with the same ESDL (same id) being submitted for OMOTES jobs many times? For instance, if the user changes ESDL a few times and then runs OMOTES jobs respectively, or if the same ESDL is shared and run by different users. Should both ESDL id and job_name, (also job run creator?) be used to identify which timeseries data to be deleted?

Also, do we need to make sure timeseries data can only be deleted by authorized users? For instance, job run creator or NWN admin

Reply:
I agree, we should you the ESDL id.
@cwang39403 Since omotes creates a new output ESDL for each job, so the ESDL id is all we need.
@cwang39403 Good point about the authorization. There is an `INFLUXDB_FRONTEND_ADMIN_USER` user with all privileges which is used by the frontend. It is up to the frontends to make sure that the right people can delete jobs.
For the mapeditor this is restricted to the creator of the job.